### PR TITLE
UI made into Rust default feature, in case you do not want x11 common library

### PR DESCRIPTION
**Pull request by Xavantex**
Changes UI dependency into a Rust feature that can be turned off at compile time.
This loses the libxkbcommon dependency, allowing most common headless environments to run binary.

Only tested to run with a test config and all the arg flags.
No hardware available currently to test on personally.

### CODE REVIEW

**Comment by Xavantex** [comment]
https://github.com/wiiznokes/fan-control/issues/138#issuecomment-2183037073

New pull request as given go ahead :)

File: src/main.rs [hunk 1]
```diff
@@ -112,10 +114,15 @@ fn try_run() -> Result<()> {
         update: Update::new(),
     };
 
-    match args.cli {
-        true => cli::run_cli(app_state),
-        false => ui::run_ui(app_state),
-    };
+    #[cfg(not(feature = "ui"))]
+    cli::run_cli(app_state)
```
Comment (wiiznokes) [inline src/main.rs:118]:
```suggestion
    cli::run_cli(app_state);
    
```

**Review by wiiznokes — CHANGES_REQUESTED** [review]
lgtm, expect the missing ';'

**Review by wiiznokes — APPROVED** [review]

File: src/main.rs [hunk 2]
```diff
@@ -112,10 +114,16 @@ fn try_run() -> Result<()> {
         update: Update::new(),
     };
 
-    match args.cli {
-        true => cli::run_cli(app_state),
-        false => ui::run_ui(app_state),
-    };
+    #[cfg(not(feature = "ui"))]
+    cli::run_cli(app_state);
+
```
Comment (wiiznokes) [inline src/main.rs:119]:
```suggestion

```

**Pull request merged**